src/pybind/mgr/cephadm/module.py

```diff
@@ -1933,7 +1933,12 @@ def update_mgrs(self, spec, services):
 
     def add_mds(self, spec):
         # type: (orchestrator.ServiceSpec) -> AsyncCompletion
-        if not spec.placement.hosts or spec.placement.count is None or len(spec.placement.hosts) < spec.placement.count:
+        spec.placement.count = spec.count 
+
+        if not spec.placement.hosts:
+            spec = NodeAssignment(spec=spec, get_hosts_func=self._get_hosts, service_type='mds').load()
+
+        if len(spec.placement.hosts) < spec.placement.count:
```

Comment:
before we all mess with this here, let's move this code to a dedicated function over here: #33041

Reply:
Have a look at #33041 !